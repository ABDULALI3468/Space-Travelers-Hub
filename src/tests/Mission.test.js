import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Mission from '../components/Mission';

import store from '../Redux/ConfigureStore';

it('renders correctly', () => {
  const mission = {
    mission_id: '1',
    mission_name: 'Test Mission',
    description: ' This is a test mission',
    reserved: true,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Mission mission={mission} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Mission component', () => {
  const mission = {
    mission_id: '1',
    mission_name: 'Test Mission',
    description: ' This is a test mission',
    reserved: true,
  };
  render(
    <Provider store={store}>
      <table>
        <tbody>
          <Mission mission={mission} />
        </tbody>
      </table>
    </Provider>,
  );
  screen.debug();
});
