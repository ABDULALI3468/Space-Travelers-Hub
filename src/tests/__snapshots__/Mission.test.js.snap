// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td
    className="bold-text mission-name"
  >
    Test Mission
  </td>
  <td
    className="description"
  >
     This is a test mission
  </td>
  <td
    className="center-content"
  >
    <span
      className="badge bg-primary"
    >
      Active Member
    </span>
  </td>
  <td
    className="center-content"
  >
    <button
      className="btn btn-outline-danger"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Leave Mission
    </button>
  </td>
</tr>
`;
